sharing/src/test/scala/io/delta/sharing/spark/DeltaFormatSharingSourceSuite.scala

```diff
@@ -0,0 +1,866 @@
+      val profileFile = prepareProfileFile(tempDir)
+      withSQLConf(getDeltaSharingClassesSQLConf.toSeq: _*) {
+        val tablePath = profileFile.getCanonicalPath + s"#share1.default.$sharedTableName"
+        val e = intercept[Exception] {
+          val df = spark.readStream
+            .format("deltaSharing")
+            .option("responseFormat", "delta")
+            .option("readChangeFeed", "true")
+            .load(tablePath)
+          testStream(df)(
+            AssertOnQuery { q =>
+              q.processAllAvailable(); true
+            }
+          )
+        }
+        assert(e.getMessage.contains("Delta sharing cdc streaming is not supported"))
+      }
+    }
+  }
+
+  test("DeltaFormatSharingSource simple query works") {
+    withTempDir { tempDir =>
+      val deltaTableName = "delta_table_simple"
+      withTable(deltaTableName) {
+        sql(s"""
+               |CREATE TABLE $deltaTableName (value STRING)
+               |USING DELTA
+               |""".stripMargin)
+
+        val sharedTableName = "shared_streaming_table_simple"
+        prepareMockedClientMetadata(deltaTableName, sharedTableName)
+
+        val profileFile = prepareProfileFile(tempDir)
+        withSQLConf(getDeltaSharingClassesSQLConf.toSeq: _*) {
+          val tablePath = profileFile.getCanonicalPath + s"#share1.default.$sharedTableName"
+
+          def InsertToDeltaTable(values: String): Unit = {
+            sql(s"INSERT INTO $deltaTableName VALUES $values")
+          }
+
+          InsertToDeltaTable("""("keep1"), ("keep2"), ("drop3")""")
+          prepareMockedClientAndFileSystemResult(deltaTableName, sharedTableName, Some(1L))
+          prepareMockedClientGetTableVersion(deltaTableName, sharedTableName)
+
+          val df = spark.readStream
+            .format("deltaSharing")
+            .option("responseFormat", "delta")
+            .load(tablePath)
+            .filter($"value" contains "keep")
+
+          testStream(df)(
+            AssertOnQuery { q =>
+              q.processAllAvailable(); true
+            },
+            CheckAnswer("keep1", "keep2"),
+            StopStream
+          )
+        }
+      }
+    }
+  }
+
+  test("restart works sharing") {
+    withTempDirs { (inputDir, outputDir, checkpointDir) =>
+      val deltaTableName = "delta_table_restart"
+      withTable(deltaTableName) {
+        createTableForStreaming(deltaTableName)
+        val sharedTableName = "shared_streaming_table_restart"
+        prepareMockedClientMetadata(deltaTableName, sharedTableName)
+        val profileFile = prepareProfileFile(inputDir)
+        val tablePath = profileFile.getCanonicalPath + s"#share1.default.$sharedTableName"
+
+        withSQLConf(getDeltaSharingClassesSQLConf.toSeq: _*) {
+          def InsertToDeltaTable(values: String): Unit = {
+            sql(s"INSERT INTO $deltaTableName VALUES $values")
+          }
+
+          def processAllAvailableInStream(): Unit = {
```

Comment:
do these tests simplified if you `testStream` (or the [streaming test framework](DeltaFormatSharingSourceSuite)) is used?